oval-schemas/independent-definitions-schema.xsd

```diff
@@ -968,6 +968,171 @@
+                              <oval:object>shellcommand_object</oval:object>
+                              <oval:state>shellcommand_state</oval:state>
+                              <oval:item target_namespace="http://oval.mitre.org/XMLSchema/oval-system-characteristics-5#independent">shellcommand_item</oval:item>
+                        </oval:element_mapping>
+                  </xsd:appinfo>
+                  <xsd:appinfo>
+                        <sch:pattern id="ind-def_shellcommand_tst">
+                              <sch:rule context="ind-def:shellcommand_test/ind-def:object">
+                                    <sch:assert test="@object_ref=ancestor::oval-def:oval_definitions/oval-def:objects/ind-def:shellcommand_object/@id"><sch:value-of select="../@id"/> - the object child element of a shellcommand_test must reference a shellcommand_object</sch:assert>
+                              </sch:rule>
+                              <sch:rule context="ind-def:shellcommand_test/ind-def:state">
+                                    <sch:assert test="@state_ref=ancestor::oval-def:oval_definitions/oval-def:states/ind-def:shellcommand_state/@id"><sch:value-of select="../@id"/> - the state child element of a shellcommand_test must reference a shellcommand_state</sch:assert>
+                              </sch:rule>
+                        </sch:pattern>
+                  </xsd:appinfo>
+            </xsd:annotation>
+            <xsd:complexType>
+                  <xsd:complexContent>
+                        <xsd:extension base="oval-def:TestType">
+                              <xsd:sequence>
+                                    <xsd:element name="object" type="oval-def:ObjectRefType" />
+                                    <xsd:element name="state" type="oval-def:StateRefType" minOccurs="0" maxOccurs="unbounded"/>
+                              </xsd:sequence>
+                        </xsd:extension>
+                  </xsd:complexContent>
+            </xsd:complexType>
+      </xsd:element>
+      
+      <xsd:element name="shellcommand_object" substitutionGroup="oval-def:object">
+            <xsd:annotation>
+                  <xsd:documentation>The shellcommand_object is used by a shellcommand_test to define a shell to use (e.g. sh, bash, ksh, etc.), a command (or shell script) to be run, and a pattern to filter result lines. The default shell is bash.  Each object extends the standard ObjectType as defined in the oval-definitions-schema and one should refer to the ObjectType description for more information. The common set element allows complex objects to be created using filters and set logic.
+                  
+The evaluation of the object should always produce at least one item.  If the object evaluation does not produce output that should result in an item, one should be created with a status of 'does not exist' and, if possible, the exit_status and any STDERR from the system call should be captured in that item.
```

Comment:
Should read: "... should always produce one item."

Reply:
... per variable value permutation, naturally, being implied.

Reply:
Confirmed.

Reply:
But you didn't change the text to say: "The evaluation of the object should always produce one item."

Reply:
Actually, I think you should rewrite it as follows:

```
The evaluation of the object should always produce one item. If the command execution does not produce output, an item should still be created with the exit_status (AKA process exit code), a stdout entity with a status of 'does not exist', and any STDERR from the execution captured in stderr_line entities.
```